mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -2549,10 +2572,15 @@ LogicalResult OperationConverter::legalizeConvertedArgumentTypes(
     });
     return liveUserIt == val.user_end() ? nullptr : *liveUserIt;
   };
-  for (auto &r : rewriterImpl.rewrites)
-    if (auto *rewrite = dyn_cast<BlockTypeConversionRewrite>(r.get()))
-      if (failed(rewrite->materializeLiveConversions(findLiveUser)))
+  // Note: `rewrites` may be reallocated as the loop is running.
+  for (int64_t i = 0; i < rewriterImpl.rewrites.size(); ++i) {
```

Comment:
Range based? (yeah seems I'm focussed on loops this morning :))

Reply:
That won't work here because things may get added to the vector while the loop is running and that could invalidate the iterator (when the vector reallocates storage).